src/features/courses/course-controller.ts

```diff
@@ -3362,6 +3363,63 @@ class CourseController {
             return result;
         });
     }
+
+    // Sends an email to the professor associated with the course.
+    async emailProfessor(options: EmailProfOptions): Promise<boolean> {
+        const coursePromise = Course.findOne({
+            where: {
+                id: options.courseId,
+            },
+            include: [{
+                model: User,
+                as: 'instructor',
+                attributes: ['id', 'firstName', 'lastName', 'email', 'preferredEmail'],
+                where: {
+                    active: true
+                },
+                required: true,
+            }],
+        });
+
+        const gradePromise = StudentGrade.findOne({
```

Comment:
@doombot-exe @red-tt-bear Do I need to also be getting the gradeInstance here, if there's one applicable? I don't remember if the randomSeed bubbles up.

Reply:
Just kidding, I'm only getting the seed from this so I don't need it at all 🤦

Reply:
This is tricky though because you need the grade instance for the student's problem number...
The problem number for the professor and the user is not necessarily the same
Good thing you included the id 😅 
Once we get nice routing we could not give any of that information and say:

Your student is having an issue with a problem. Click this link for more information

One other thing that could be looked at as a pro or a con by not giving extra information in the email is that it never gets stale, reordering the question won't matter (though I don't think this is a big deal)